# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../blackbox/gcc43.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN blackbox-0.70.1-orig/lib/Image.cc blackbox-0.70.1/lib/Image.cc
--- blackbox-0.70.1-orig/lib/Image.cc	2005-04-08 17:41:09.000000000 +0200
+++ blackbox-0.70.1/lib/Image.cc	2010-08-13 19:34:25.404254758 +0200
@@ -27,6 +27,8 @@
 #include "Pen.hh"
 #include "Texture.hh"
 
+#include <string.h>
+
 #include <algorithm>
 #include <vector>
 
diff -ruN blackbox-0.70.1-orig/lib/Resource.cc blackbox-0.70.1/lib/Resource.cc
--- blackbox-0.70.1-orig/lib/Resource.cc	2005-04-06 23:16:50.000000000 +0200
+++ blackbox-0.70.1/lib/Resource.cc	2010-08-13 19:34:25.408254575 +0200
@@ -29,6 +29,7 @@
 #include <X11/Xresource.h>
 
 #include <stdio.h>
+#include <string.h>
 
 
 bt::Resource::Resource(void)
diff -ruN blackbox-0.70.1-orig/lib/XDG.cc blackbox-0.70.1/lib/XDG.cc
--- blackbox-0.70.1-orig/lib/XDG.cc	2005-04-06 16:04:38.000000000 +0200
+++ blackbox-0.70.1/lib/XDG.cc	2010-08-13 19:34:25.408254575 +0200
@@ -27,6 +27,8 @@
 
 #include <stdlib.h>
 
+#include <algorithm>
+
 
 // make sure directory names end with a slash
 static std::string terminateDir(const std::string &string)
diff -ruN blackbox-0.70.1-orig/src/BlackboxResource.cc blackbox-0.70.1/src/BlackboxResource.cc
--- blackbox-0.70.1-orig/src/BlackboxResource.cc	2005-10-18 09:34:46.000000000 +0200
+++ blackbox-0.70.1/src/BlackboxResource.cc	2010-08-13 19:34:25.408254575 +0200
@@ -33,6 +33,7 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
+#include <string.h>
 
 BlackboxResource::BlackboxResource(const std::string& rc): rc_file(rc) {
   screen_resources = 0;
diff -ruN blackbox-0.70.1-orig/src/Screen.cc blackbox-0.70.1/src/Screen.cc
--- blackbox-0.70.1-orig/src/Screen.cc	2005-10-18 10:07:22.000000000 +0200
+++ blackbox-0.70.1/src/Screen.cc	2010-08-13 19:34:25.404254758 +0200
@@ -48,6 +48,8 @@
 #include <ctype.h>
 #include <dirent.h>
 
+#include <string.h>
+
 
 static bool running = true;
 static int anotherWMRunning(Display *, XErrorEvent *) {
diff -ruN blackbox-0.70.1-orig/src/ScreenResource.cc blackbox-0.70.1/src/ScreenResource.cc
--- blackbox-0.70.1-orig/src/ScreenResource.cc	2005-04-13 07:54:08.000000000 +0200
+++ blackbox-0.70.1/src/ScreenResource.cc	2010-08-13 19:34:25.408254575 +0200
@@ -32,7 +32,7 @@
 #include <Resource.hh>
 
 #include <assert.h>
-
+#include <string.h>
 
 static const int iconify_width  = 9;
 static const int iconify_height = 9;
diff -ruN blackbox-0.70.1-orig/src/main.cc blackbox-0.70.1/src/main.cc
--- blackbox-0.70.1-orig/src/main.cc	2005-01-03 10:42:57.000000000 +0100
+++ blackbox-0.70.1/src/main.cc	2010-08-13 19:34:25.408254575 +0200
@@ -35,6 +35,7 @@
 #include "../version.h"
 
 #include <stdio.h>
+#include <string.h>
 
 
 static void showHelp(int exitval) {
diff -ruN blackbox-0.70.1-orig/util/bsetroot.cc blackbox-0.70.1/util/bsetroot.cc
--- blackbox-0.70.1-orig/util/bsetroot.cc	2005-03-15 08:01:37.000000000 +0100
+++ blackbox-0.70.1/util/bsetroot.cc	2010-08-13 19:34:25.408254575 +0200
@@ -30,7 +30,10 @@
 #include <cctype>
 
 #include <X11/Xatom.h>
+
+#include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 
 // ignore all X errors
